linux/fedora-rpm/audacity.spec

```diff
@@ -0,0 +1,857 @@
+    -Daudacity_lib_preference=system \
+    -Daudacity_use_sqlite=system \
+    -Daudacity_use_sndfile=system \
+    -Daudacity_use_soxr=system \
+    -Daudacity_use_lame=system \
+%if 0%{?rhel} == 8
+    -Daudacity_use_twolame=off \
+%else
+    -Daudacity_use_twolame=system \
+%endif
+    -Daudacity_use_flac=system \
+    -Daudacity_use_ladspa=on \
+    -Daudacity_use_vorbis=system \
+    -Daudacity_use_id3tag=system \
+    -Daudacity_use_expat=system \
+    -Daudacity_use_soundtouch=system \
+    -Daudacity_use_vamp=system \
+    -Daudacity_use_lv2=system \
+    -Daudacity_use_portaudio=local \
+    -Daudacity_use_midi=system \
+    -Daudacity_use_ogg=system \
+%if %{with ffmpeg}
+%if ! %{with local_ffmpeg}
+    -Daudacity_use_ffmpeg=loaded \
+%endif
+%else
+    -Daudacity_use_ffmpeg=off \
+%endif
+
+%cmake_build
+
+
+%install
+%if (0%{?fedora} && 0%{?fedora} < 33)
+cd build
+%endif
+
+%cmake_install
+
+%if (0%{?fedora} && 0%{?fedora} < 33)
+cd -
+%endif
+
+rm -Rf $RPM_BUILD_ROOT%{_datadir}/%{name}/include
+
+# Remove a helper script, that runs audacity in GitHub CI builds
+rm -f $RPM_BUILD_ROOT/usr/%{name}
+
+%if 0%{?rhel} >= 8 || 0%{?fedora}
+if appstream-util --help | grep -q replace-screenshots ; then
+# Update the screenshot shown in the software center
+#
+# NOTE: It would be *awesome* if this file was pushed upstream.
+#
+# See http://people.freedesktop.org/~hughsient/appdata/#screenshots for more details.
+#
+appstream-util replace-screenshots $RPM_BUILD_ROOT%{_datadir}/metainfo/audacity.appdata.xml \
+  https://raw.githubusercontent.com/hughsie/fedora-appstream/master/screenshots-extra/audacity/a.png
+fi
+%endif
+
+mkdir -p $RPM_BUILD_ROOT%{_datadir}/%{name}/help/manual
+# audacity manual must be unzipped to correct location
+unzip %{SOURCE1} -d $RPM_BUILD_ROOT%{_datadir}/%{name}/help
+
+
+%{find_lang} %{name}
+
+desktop-file-install --dir $RPM_BUILD_ROOT%{_datadir}/applications \
+        $RPM_BUILD_ROOT%{_datadir}/applications/audacity.desktop
+
+mkdir %{buildroot}%{_datadir}/doc/%{name}/nyquist
+cp -pr lib-src/libnyquist/nyquist/license.txt %{buildroot}%{_datadir}/doc/%{name}/nyquist
+cp -pr lib-src/libnyquist/nyquist/Readme.txt %{buildroot}%{_datadir}/doc/%{name}/nyquist
+rm %{buildroot}%{_datadir}/doc/%{name}/LICENSE.txt
+
+%files -f %{name}.lang
+%{_bindir}/%{name}
+%{_libdir}/%{name}
+%dir %{_datadir}/%{name}
+%{_datadir}/%{name}/EQDefaultCurves.xml
+# Modules go to lib/audacity
+#{_datadir}/%{name}/modules/
+%{_datadir}/%{name}/nyquist/
+%{_datadir}/%{name}/plug-ins/
+%exclude %{_datadir}/%{name}/help
+%{_mandir}/man*/*
+%{_datadir}/applications/*
+%{_datadir}/metainfo/%{name}.appdata.xml
+%{_datadir}/pixmaps/*
+%{_datadir}/icons/hicolor/*/%{name}.png
+%{_datadir}/icons/hicolor/scalable/apps/%{name}.svg
+%{_datadir}/mime/packages/*
+%{_datadir}/doc/%{name}
+%license LICENSE.txt
+
+%files manual
+%dir %{_datadir}/%{name}
+%dir %{_datadir}/%{name}/help
+%{_datadir}/%{name}/help/manual/
+
+
+%changelog
```

Comment:
I would remove %changelog here , it is not required to build the package, ut this %changelog will become obsolete quickly because this spec will be maintained separately.

Also I would add %check section like this:

```bash
%check
p="$(patchelf --print-needed %buildroot/%_bindir/audacity)"
# upstream seems to assume statically linking bundled libsbsms,
# verify that system one is used
echo "$p" | grep -q sbsms
# ffmpeg and mp3lame can be either dlopen'ed or linked explicitly,
# ensure that they are linked explicitly
echo "$p" | grep -q libavcodec
echo "$p" | grep -q libmp3lame
```

(BuildRequires: patchelf)